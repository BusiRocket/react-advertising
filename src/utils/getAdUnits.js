import getPrebidSizeMapping from './getPrebidSizeMapping';

export default (slots, sizeMappings) =>
    slots.reduce(
        (acc, currSlot) =>
            acc.concat(
                currSlot.prebid.map(currPrebid => {
                    const adUnit = {
                        code: currSlot.id,
                        sizes: currPrebid.sizes,
                        bids: currPrebid.bids
                    };

                    const sizeMapping = getPrebidSizeMapping(currSlot.sizeMappingName, sizeMappings);
                    if (sizeMapping) {
                        adUnit.sizeMapping = sizeMapping;
                    }

                    return adUnit;
                })
            ),
        []
    );
