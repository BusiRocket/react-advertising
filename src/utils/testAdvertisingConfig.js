export const DIV_ID_FOO = 'div-gpt-ad-foo';
const SLOT_ID_FOO = 666;
const PREBID_SIZES_FOO = [[320, 240]];
const BIDDER_FOO = 'bidder-foo';
const PARAMS_FOO = { foo: 'bar' };

export const DIV_ID_BAR = 'div-gpt-ad-bar';
const SLOT_ID_BAR = [11, 666];
const PREBID_SIZES_BAR = [[320, 240]];
const BIDDER_BAR = 'bidder-bar';
const PARAMS_BAR = { bar: 'foo' };
const COLLAPSE_EMPTY_DIV = [true, true];

const TIMEOUT = 5000;
const GPT_SIZES = ['fluid', [320, 240]];

const GLOBAL_AD_UNIT_PATH = 'global/ad/unit/path';
const SLOT_AD_UNIT_PATH = 'slot/ad/unit/path';
const USD_TO_EUR_RATE = 2;
const MAKE = 'make';
const MODELS = 'models';
const LOGGED_IN = false;
const THREAD_ID = 6666;
const PLACEMENT_TEST_ID = 100;

export const config = {
    active: true,
    prebid: {
        timeout: TIMEOUT
    },
    slots: [
        {
            id: DIV_ID_FOO,
            targeting: { a: SLOT_ID_FOO },
            sizes: GPT_SIZES,
            prebid: [
                {
                    sizes: PREBID_SIZES_FOO,
                    bids: [
                        {
                            bidder: BIDDER_FOO,
                            params: PARAMS_FOO
                        }
                    ]
                }
            ]
        },
        {
            id: DIV_ID_BAR,
            adUnitPath: SLOT_AD_UNIT_PATH,
            targeting: { a: SLOT_ID_BAR },
            sizes: GPT_SIZES,
            sizeMappingName: 'mobailAndTablet',
            collapseEmptyDiv: COLLAPSE_EMPTY_DIV,
            prebid: [
                {
                    sizes: PREBID_SIZES_BAR,
                    bids: [
                        {
                            bidder: BIDDER_BAR,
                            params: PARAMS_BAR
                        }
                    ]
                }
            ]
        }
    ],
    sizeMappings: {
        mobailAndTablet: [
            {
                viewPortSize: [0, 0],
                sizes: []
            },
            {
                viewPortSize: [320, 700],
                sizes: [[300, 250], [320, 50]]
            },
            {
                viewPortSize: [1050, 200],
                sizes: []
            }
        ]
    },
    metaData: {
        adUnitPath: {
            path: GLOBAL_AD_UNIT_PATH
        },
        boardMakeAndModels: [
            {
                make: MAKE,
                models: MODELS
            }
        ],
        usdToEurRate: USD_TO_EUR_RATE,
        loggedIn: LOGGED_IN,
        threadId: THREAD_ID
    },
    placementTestId: PLACEMENT_TEST_ID
};
