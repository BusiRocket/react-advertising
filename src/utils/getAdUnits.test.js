import getAdUnits from './getAdUnits';
import { config } from './testAdvertisingConfig';

describe('[@mt-advertising/main/utils/getAdUnits]', () =>
    describe('When I get ad units from a given config', () => {
        describe('without size mappings', () => {
            let adUnits;
            beforeEach(() => (adUnits = getAdUnits(config.slot)));
            describe('the result', () => it('is correct', () => expect(adUnits).toMatchSnapshot()));
        });
        describe('with size mappings', () => {
            let adUnits;
            beforeEach(() => (adUnits = getAdUnits(config.slot, config.sizeMappings)));
            describe('the result', () => it('is correct', () => expect(adUnits).toMatchSnapshot()));
        });
    }));
