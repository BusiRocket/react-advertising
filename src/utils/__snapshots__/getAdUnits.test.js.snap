// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I get ad units from a given config with size mappings the result is correct 1`] = `
Array [
  Object {
    "bids": Array [
      Object {
        "bidder": "bidder-foo",
        "params": Object {
          "foo": "bar",
        },
      },
    ],
    "code": "div-gpt-ad-foo",
    "sizes": Array [
      Array [
        320,
        240,
      ],
    ],
  },
  Object {
    "bids": Array [
      Object {
        "bidder": "bidder-bar",
        "params": Object {
          "bar": "foo",
        },
      },
    ],
    "code": "div-gpt-ad-bar",
    "sizeMapping": Array [
      Object {
        "minWidth": 1050,
        "sizes": Array [],
      },
      Object {
        "minWidth": 320,
        "sizes": Array [
          Array [
            300,
            250,
          ],
          Array [
            320,
            50,
          ],
        ],
      },
      Object {
        "minWidth": 0,
        "sizes": Array [],
      },
    ],
    "sizes": Array [
      Array [
        320,
        240,
      ],
    ],
  },
]
`;

exports[`When I get ad units from a given config without size mappings the result is correct 1`] = `
Array [
  Object {
    "bids": Array [
      Object {
        "bidder": "bidder-foo",
        "params": Object {
          "foo": "bar",
        },
      },
    ],
    "code": "div-gpt-ad-foo",
    "sizes": Array [
      Array [
        320,
        240,
      ],
    ],
  },
  Object {
    "bids": Array [
      Object {
        "bidder": "bidder-bar",
        "params": Object {
          "bar": "foo",
        },
      },
    ],
    "code": "div-gpt-ad-bar",
    "sizes": Array [
      Array [
        320,
        240,
      ],
    ],
  },
]
`;
